packages/kbn-monaco/src/xjson/grammar.ts

```diff
@@ -96,37 +83,42 @@ export const createParser = () => {
           (string += ch), next();
       return (number = +string), isNaN(number) ? (error('Bad number'), void 0) : number;
     },
+    stringLiteral = function(quotes: string) {
```

Comment:
The original functionality was airlifted out of:

https://github.com/elastic/kibana/blob/854e0d4d850f160359d8a923a0b2128f0b0ad8a6/packages/kbn-ace/src/ace/modes/x_json/worker/x_json.ace.worker.js#L797

This is, unfortunately, not exactly the most reader-friendly code and so without tests it's all the more tough to say if a regression has snuck in or not. I know with it being "the same-ish" as Console we had more guarantees that the code worked as expected (according to the tests on Console). I apologise for not adding tests here at the time 🤦🏻.

I think it would be useful to add tests for the difference you're expecting to see with these changes, but AFAICT it looks like this was just a refactor and we are expecting the same behaviour? If that's the case I'd recommend breaking this change out into a follow up PR that adds tests for the "before" and "after" behaviour against this parser.

Let me know what you think, or whether I've missed something!

Reply:
> I think it would be useful to add tests for the difference you're expecting to see with these changes, but AFAICT it looks like this was just a refactor and we are expecting the same behaviour? If that's the case I'd recommend breaking this change out into a follow up PR that adds tests for the "before" and "after" behaviour against this parser.

++ Great suggestion. I'd also feel more comfortable with this change if I better understood the difference in behavior we're expecting (if any) to help ensure we're not introducing any regressions.